Cython/Utility/CppConvert.pyx

```diff
@@ -235,3 +235,24 @@ cdef object {{cname}}(const std_complex[X]& z):
     tmp.real = <double>z.real()
     tmp.imag = <double>z.imag()
     return tmp
+
+#################### enum_class.from_py ####################
+
+cdef extern from *:
+    cdef enum class enumtype "{{type}}":
+        pass
```

Comment:
Why is this needed? Can't we just use `{{type}}` as cast and return type below?

Reply:
I struggled a lot with getting that to work. Below a summary; perhaps I was missing something obvious:

The problem is essentially making `{{type}}` visible to the utility code. Looking at other examples in the code base, I figured that the `outer_module_scope=` arg here would take care of that:

```
        env.use_utility_code(CythonUtilityCode.load(
            "enum_class.to_py",
            "CppConvert.pyx",
            context=context,
            outer_module_scope=env.global_scope(),  # need access to types declared in module
            compiler_directives=env.directives
        ))
```

However, it fails when the type hasn't been `cimport'd` directly, but rather accessed with `dotted` notation:

```
from cheese cimport Cheese
cdef Cheese d = int(1)  # this worked fine
```

```
cimport cheese
cdef cheese.Cheese d = int(1)  # this didn't work 
```

Then, I tried adding the type to the context so that it would be picked up [`context_types`](https://github.com/cython/cython/blob/master/Cython/Compiler/UtilityCode.py#L79). That resulted in breaking the symbol table entry for the enum. This is where I struggled; I eventually traced it to https://github.com/cython/cython/blob/master/Cython/Compiler/UtilityCode.py#L177, where the entry gets overwritten, and loses its `enum_values`, breaking things downstream.

Reply:
This is no longer relevant as I removed this utility code :)